trieste/bayesian_optimizer.py

```diff
@@ -493,9 +590,16 @@ def optimize(
             try:
 
                 if track_state:
-                    models_copy = copy.deepcopy(models)
-                    acquisition_state_copy = copy.deepcopy(acquisition_state)
-                    history.append(Record(datasets, models_copy, acquisition_state_copy))
+                    if track_path is None:
+                        datasets_copy = copy.deepcopy(datasets)
+                        models_copy = copy.deepcopy(models)
+                        acquisition_state_copy = copy.deepcopy(acquisition_state)
+                        history.append(Record(datasets_copy, models_copy, acquisition_state_copy))
+                    else:
+                        track_path = Path(track_path)
+                        record = Record(datasets, models, acquisition_state)
+                        record_path = track_path / f"step.{step:0{len(str(num_steps-1))}d}.pickle"
```

Comment:
perhaps avoiding hardcoding here, filename "step" could be defined as a constant at the top of the file to make it changeable and can then be reused in other methods